tests/functional/backend/XssAndScriptCheckTest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace ls\tests;
+
+/**
+ * @group api
+ */
+class XssAndScriptCheckTest extends TestBaseClass
+{
+    /**
+     * @var integer $userId
+     */
+    private static $userId;
+
+    /**
+     * @inheritdoc
+     */
+    public static function setUpBeforeClass(): void
+    {
+        parent::setupBeforeClass();
+        /* Create an random user and login */
+        $username = "test_" . \Yii::app()->securityManager->generateRandomString(8);
+        $password = createPassword();
+        self::$userId = \User::insertUser($username, $password, 'Test user for XSS', 1, 'user@example.org');
+        \Permission::model()->setGlobalPermission(self::$userId, 'surveys', array('create_p'));
+        \Permission::model()->setGlobalPermission(self::$userId, 'auth_db', array('read_p'));
+    }
+
+    /**
+     * @return void
+     */
+    public function testXssOnAndScriptOn()
+    {
+        App()->setConfig('filterxsshtml', true);
+        App()->setConfig('disablescriptwithxss', true);
+        $questionl10n = $this->importSurveyAndgetQuestionI10n();
```

Comment:
So you are testing the validator by triggering an import.
Can't you just do a question save? Isn't that a more direct test?

Also, maybe a unit test to isScriptUpdateAllowed method?

Reply:
>  Can't you just do a question save? Isn't that a more direct test?

It must be done too in my opinion, but import **must** work too (and in general is the 2st place where it's not done : only usage of model, in GUI : we use controller).

Like it must be done on group, survey content etc …

> Also, maybe a unit test to isScriptUpdateAllowed method could be usefull?

Script update is allowed if XSS is POFF. The 2 feature are really linked. Test both at same time seems better.

Reply:
> Also, maybe a unit test to isScriptUpdateAllowed method could be usefull?

better understand : `testXssOnAndScriptOff` do it